### PR TITLE
observe and deal with map-style mutations

### LINKED ISSUE #933: Create a mutation observer for the content of map-style
When the content of the \<map-style> changes, it could propagate those changes to the \<style> element to which it's linked (`.styleElement`).   There are probably other advances that could be made with constructed stylesheets and so on, tbd, but it would be nice for the style of rendered features to update in real time with changes to the stylesheet contents.

### PR DESCRIPTION
**Pull request by AliyanH**
resolves #933

### CODE REVIEW

**Comment by prushforth** [comment]
Quite a few problems in experiments with a build of this branch... though the tests seem mostly clean. Strange!

File: src/map-style.js [hunk 1]
```diff
@@ -40,6 +40,16 @@ export class MapStyle extends HTMLElement {
         target.setAttribute(attribute.nodeName, attribute.nodeValue);
       });
     }
+
+    // use observer to monitor the changes in mapStyle textContent
+    this._observer = new MutationObserver(() => {
+      this.styleElement.textContent = this.textContent;
+    });
+    this._observer.observe(this, {
```
Comment (prushforth) [inline src/map-style.js:48]: I wonder if we only need to monitor the `characterData`; there shouldn't be a subtree or child nodes except textContent

Comment (prushforth) [reply src/map-style.js:48]: I tried removing the `childList` and `subtree`, then just `subtree`, and the test below failed both times, so I guess the answer is : no ;-P

**Review by prushforth — APPROVED** [review]
LGTM

File: src/map-style.js (hunk 1 above)
Comment (AliyanH) [reply src/map-style.js:48]: Yeah it was pretty odd, I played around also, and it appeared these three were required. Atleast I was able to remove `attributes: true, attributeOldValue: true` which we use in map-feature. Perhaps they can also be removed from map-feature.

**Pull request merged**